tests/Admin/Admin_Page_Tests.php

```diff
@@ -39,6 +39,7 @@ public function test_add_page() {
 		wp_set_current_user( $admin_user );
 		set_current_screen( 'dashboard' );
 
+		$this->admin_page->add_and_initialize_page();
 		$page_hook    = $this->admin_page->add_page();
```

Comment:
See my previous comment, you're still calling both of these methods in a single test, which isn't right.

I mention above that the `$hook_suffix` class property isn't used, but you could make use of it here which would justify its existence more. Instead of calling `add_page()`, use reflection on the `Admin_Page` instance to get the value of the property.

Reply:
`$hook_suffix` is used 😕 In `add_page`. I added this class property so we can call `add_page` again here without it run the same code twice. I could maybe convert this to a class reflection to get the class property, but that seems a bit like overkill.

Reply:
See my above reply, using the class property just to avoid duplicate execution because of a test is not a good pattern. Using reflection to get the class property is more appropriate as that way this test works without an incorrect logic flow.

I think instead of line 43 here we should get the value from `Admin_Page::$hook_suffix`, and its usage from the `add_page()` method should be removed.

Reply:
The property is protected. So it isn't that easy.